### PR TITLE
AAE-18301 new action to clean GH cache after pr merge

### PR DESCRIPTION
**Pull request by wojciech-piotrowiak**
### Checklist

- Jira Reference (also in PR title):
- [README](https://github.com/Alfresco/alfresco-build-tools/blob/master/docs/README.md) updated after adding/changing behaviour of an action
- Proposed version increment for [release](https://github.com/Alfresco/alfresco-build-tools/blob/master/docs/README.md#release):
  - [ ] Patch (bugfix)
  - [X] Minor (new feature)
  - [ ] Major (breaking changes)
- External PR link where changes has been tested:

### CODE REVIEW

File: .github/actions/gh-cache-cleanup-on-merge/action.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: "GH cache cleanup"
+description: "Performs the cleanup of cache entries related with already closed PR"
```
Comment (wojciech-piotrowiak) [inline .github/actions/gh-cache-cleanup-on-merge/action.yml:2]: source: https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#force-deleting-cache-entries

Comment (atchertchian) [reply .github/actions/gh-cache-cleanup-on-merge/action.yml:2]: Please document this new action in the README, with a sample usage workflow

Comment (wojciech-piotrowiak) [reply .github/actions/gh-cache-cleanup-on-merge/action.yml:2]: Done

File: docs/README.md [hunk 2]
```diff
@@ -1286,6 +1286,16 @@ Runs Veracode Source Clear Scan
           srcclr-install-options: '-DskipTestModules' # optional, additional maven options
 ```
 
+### github cache cleanup
+
+Performs the cleanup of cache entries related with already closed PR
+
+```yaml
+      - uses: Alfresco/alfresco-build-tools/.github/actions/gh-cache-cleanup-on-merge@ref
```
Comment (atchertchian) [inline docs/README.md:1294]:
```suggestion
name: Cleanup caches for work branch
on:
  pull_request:
    types:
      - closed

jobs:
  cleanup:
    runs-on: ubuntu-latest
    steps:
      - uses: Alfresco/alfresco-build-tools/.github/actions/gh-cache-cleanup-on-merge@ref
```

Comment (atchertchian) [reply docs/README.md:1294]: (suggested to add the even triggering the workflow for better guidance)

**Review by gionn — APPROVED** [review]
looks cool.

it would be great to have a script which monitor the cache usages across the org repos and reports where adding this workflow could help

File: version.txt [hunk 3]
```diff
@@ -1 +1 @@
-v3.8.1
+v3.9.1
```
Comment (gionn) [inline version.txt:1]:
```suggestion
v3.9.0
```

**Review by gionn — CHANGES_REQUESTED** [review]

**Review by gionn — APPROVED** [review]

**Pull request merged**